src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -121,6 +124,17 @@ object DepositHandler {
     }
   }
 
+  def checkBaseDir()(implicit id: String, baseDir: File): Try[Unit] = Try {
```

Comment:
One idea might be to give `FailedDepositException` only when there is something wrong on our side.  And when the cause is in the deposited bag, then always give `InvalidDepositException`.  And when `FailedDepositException`, then not to send the cause to the customer, but some other kind of explanation.